### ISSUE TITLE
Navigation Route Options Issues

### ISSUE THREAD

**Issue by aw1875** [mapbox/mapbox-navigation-ios#3615]
My project is meant to work similar to the [Advanced Navigation App](https://docs.mapbox.com/ios/navigation/examples/advanced/) from the documentation but I am having some issues. Before instantiating my MapboxNavigationService I am setting some route options up but they don't seem to be working correctly. My code looks like this:

```swift
guard let navigationRouteOptions = navigationRouteOptions else { return }

// Set Route Options
navigationRouteOptions.roadClassesToAvoid = .toll
navigationRouteOptions.profileIdentifier = .automobileAvoidingTraffic
navigationRouteOptions.distanceMeasurementSystem = .metric

// Instantiate Navigation Service
let navigationService = MapboxNavigationService(routeResponse: routeResponse, routeIndex: currentRouteIndex, routeOptions: navigationRouteOptions, simulating: .onPoorGPS)
```
However, when I run the app and test the code, none of the route options I set are applied and I can't figure out why. Any help would be greatly appreciated.

Thanks!

**Comment by S2Ler** [comment]
👋 @aw1875, you must to set these parameters before your request the route. In the linked [Advanced Navigation App](https://docs.mapbox.com/ios/navigation/examples/advanced/) it is done in `func requestRoute(destination: CLLocationCoordinate2D)` method.

**Comment by aw1875** [comment]
@S2Ler Thanks! I guess I was calling it before the actual Navigation View loaded instead of during the function that generates the route. It seems like I was able to get `roadClassesToAvoid` and `profileIdentifier` to work but the `distanceMeasurementSystem` is still stuck with imperial system for some reason.

**Comment by S2Ler** [comment]
> @S2Ler Thanks! I guess I was calling it before the actual Navigation View loaded instead of during the function that generates the route. It seems like I was able to get `roadClassesToAvoid` and `profileIdentifier` to work but the `distanceMeasurementSystem` is still stuck with imperial system for some reason.

Yeah, it is a known bug which has been fixed already in the upcoming v2.1.

**Comment by aw1875** [comment]
Cheers, thank you for being so helpful and proactive with issues!
